src/library/scala/util/Properties.scala

```diff
@@ -168,27 +168,31 @@ private[scala] trait PropertiesTrait {
 
   /** Compares the given specification version to the specification version of the platform.
    *
-   * @param version a specification version of the form "major.minor"
+   * @param version a specification version number (legacy forms acceptable)
    * @return `true` iff the specification version of the current runtime
    * is equal to or higher than the version denoted by the given string.
    * @throws NumberFormatException if the given string is not a version string
    *
    * @example {{{
-   * // In this example, the runtime's Java specification is assumed to be at version 1.7.
+   * // In this example, the runtime's Java specification is assumed to be at version 8.
    * isJavaAtLeast("1.6")            // true
-   * isJavaAtLeast("1.7")            // true
-   * isJavaAtLeast("1.8")            // false
+   * isJavaAtLeast("1.8")            // true
+   * isJavaAtLeast("8")              // true
+   * isJavaAtLeast("9")              // false
+   * isJavaAtLeast("1.9")            // throws
    * }}}
    */
   def isJavaAtLeast(version: String): Boolean = {
-    def parts(x: String) = {
-      val i = x.indexOf('.')
-      if (i < 0) throw new NumberFormatException("Not a version: " + x)
-      (x.substring(0, i), x.substring(i+1, x.length))
+    def versionOf(s: String): Int = s.indexOf('.') match {
```

Comment:
Actually, re-reading what they intend by $MINOR, which includes revision of API, it's possible this API should support $MAJOR.$MINOR. I know I care if we're on java 8 or 9, but will I care if it is 9.1 or 9.2? assuming the spec string is 9.1. Alternatively, maybe supply `isJavaMajorSpec` for platforms that don't want to use regex for this test; otherwise just use regex like normal people.